### PR TITLE
first version of policy descriptor

### PR DESCRIPTION
**Pull request by efotopoulou**

### CODE REVIEW

**Comment by mpeuster** [comment]
I suggest to add a test to automatically validate the policy example against the policy schema.

Like done here:
- https://github.com/sonata-nfv/tng-schema/blob/master/package-specification/test/test_5gtango-ns-package-example.sh
- https://github.com/sonata-nfv/tng-schema/blob/master/service-descriptor/test/test_simplest-example.sh

**Review by mpeuster — COMMENTED** [review]
Schema as such looks fine to me.

**Review by mpeuster — CHANGES_REQUESTED** [review]
I suggest to add a test to automatically validate the policy example against the policy schema.

Like done here:
- https://github.com/sonata-nfv/tng-schema/blob/master/package-specification/test/test_5gtango-ns-package-example.sh
- https://github.com/sonata-nfv/tng-schema/blob/master/service-descriptor/test/test_simplest-example.sh

File: policy-descriptor/examples/policy-example.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# do NOT delete: this is used by tng-sdk-descriptorgen
```
Comment (StefanUPB) [inline policy-descriptor/examples/policy-example.yml]: you can delete these two comments as they only apply to the default VNFD and NSD at the moment

File: policy-descriptor/policy-schema.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    properties:
+      name:
+       description: "The name of the parameter to monitor. The name has to be supported by the service platform or the FSM."
+       type: "string"
+       pattern: "^[A-Za-z-_]+$"
+      unit:
+       description: "The unit used to monitor (or represent) the parameter."
+       oneOf:
+         - $ref: "#/definitions/general_units"
+         - $ref: "#/definitions/memory_units"
```
Comment (stefanbschneider) [inline policy-descriptor/policy-schema.yml]: I guess, you need to include these definitions at the top too in order to reference them

File: policy-descriptor/examples/policy-example.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+descriptor_schema: "https://raw.githubusercontent.com/sonata-nfv/tng-schema/master/policy-descriptor/policy-schema.yml"
+
+name: "samplePolicy"
+
+
+rules:
+  - name: "FirstRule"
+    duration: "to be defined"
+    condition: "to be defined"
+    action: "to be defined"
```
Comment (stefanbschneider) [inline policy-descriptor/examples/policy-example.yml]: In the schema, you define action to be an array of objects with mandatory `type` field. I guess, you should set up tests (as Manuel suggested) in order to ensure the descriptors match the schema (and vice versa).

File: policy-descriptor/examples/policy-example.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# do NOT delete: this is used by tng-sdk-descriptorgen
+# default YAML description of an ubuntu vm
```
Comment (stefanbschneider) [inline policy-descriptor/examples/policy-example.yml]: you can delete these two comments as they only apply to the default VNFD and NSD at the moment

**Review by stefanbschneider — CHANGES_REQUESTED** [review]
Nice work :) But I suggest, you add tests and make sure the descriptor matches the defined schema.
You can check the other descriptors to see how the tests are structured and of course, I'm happy to help.

File: policy-descriptor/examples/.~lock.policy-example.yml# [hunk 5]
```diff
@@ -0,0 +1 @@
+,eleni,elenipc.home,23.02.2018 15:45,file:///home/eleni/.config/libreoffice/4;
```
Comment (mpeuster) [inline policy-descriptor/examples/.~lock.policy-example.yml#]: this is a temporary local file that should be removed from the PR.

**Comment by stefanbschneider** [comment]
Thanks for adding the tests. Now just remove the local files:
https://github.com/ubitech/tng-schema/blob/master/policy-descriptor/examples/.directory
https://github.com/ubitech/tng-schema/blob/master/policy-descriptor/examples/.%7Elock.policy-example.yml%23
Then the PR is ready to merge :)

**Review by stefanbschneider — APPROVED** [review]

**Review by mpeuster — APPROVED** [review]

**Pull request merged**